extension/encoding/jsonlogencodingextension/json_test.go

```diff
@@ -12,20 +14,78 @@ import (
 
 func TestMarshalUnmarshal(t *testing.T) {
 	t.Parallel()
-	e := &jsonLogExtension{
-		config: &Config{
-			Mode: JSONEncodingModeBody,
+
+	tests := []struct {
+		name         string
+		decodingMode ProcessingMode
+		input        string
+		wantLogs     int
```

Comment:
For testing, the [golden](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/golden) package is great because then you can compare the whole log structure instead of just comparing the length. For example, [like this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0561223b81add994d36414c375ccc7a5a473b6b5/extension/encoding/awslogsencodingextension/internal/unmarshaler/s3-access-log/unmarshaler_test.go#L222-L224).

Reply:
Added this with commit https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40932/commits/7a80f2da52b49ecc727013d7ce35859ab56e3b10 :)